hypothesis-python/src/hypothesis/extra/array_api.py

```diff
@@ -760,10 +842,17 @@ def indices(
     )
 
 
-def make_strategies_namespace(xp: Any) -> SimpleNamespace:
+# Cache for make_strategies_namespace()
+_args_to_xps = WeakValueDictionary()
+
+
+def make_strategies_namespace(
+    xp: Any, *, api_version: Union["Ellipsis", NominalVersion] = ...
```

Comment:
Oops.  I know I asked for `...` here, but [the array api spec` uses `None` to mean "whatever is latest"](https://data-apis.org/array-api/latest/API_specification/generated/signatures.array_object.array.__array_namespace__.html), and I think we should do the same - consistency with the specific upstream convention over Hypothesis' general convention. 
 Sorry!

Reply:
Heh my thinking was you wanted to distinguish from the default `None`, as technically we're not passing `api_version=None` to `x.__array_namespace__` (which eh could be a good idea, but might be too much? don't have a good sense for default kwarg values heh). I could of clarified :sweat_smile: easy change anywho

Reply:
Gah.  Now if the array api library supports a newer version than Hypothesis does (more plausible if the user hasn't updated Hypothesis, of course, but that does happen), we'll resolve `None` to a different API version 😬 

I think the correct way around this is to get the array api namespace (`xp`) using `api_version=None`, then [check the `.__array_api_version__ ` attribute](https://data-apis.org/array-api/latest/future_API_evolution.html#versioning) and error if it's not one that we know about.  Thoughts?  This would also replace our current loop over versions.

Reply:
> we'll resolve `None` to a different API version

Ayup :upside_down_face:

> I think the correct way around this is to get the array api namespace (`xp`) using `api_version=None`, then [check the `.__array_api_version__ ` attribute](https://data-apis.org/array-api/latest/future_API_evolution.html#versioning) and error if it's not one that we know about. Thoughts? This would also replace our current loop over versions.

... I completely missed this was in the spec heh. Yep makes sense, just no one (well NumPy/CuPy/Dask) has implemented this, seemingly as it was introduced ~2yrs ago and folks forgot about it before development of `numpy.array_api` started. I'll get a feeler for this tomorrow, but my impression is it's a-okay to assume and I'll just have to write issues/PRs for the existing implementations as a follow-up.

Reply:
Also an issue for the conformance test suite then?